src/ec2/model.ts

```diff
@@ -4,29 +4,102 @@
  */
 import * as vscode from 'vscode'
 import { Session } from 'aws-sdk/clients/ssm'
+import { AWSError, IAM } from 'aws-sdk'
 import { Ec2Selection } from './utils'
 import { getOrInstallCli } from '../shared/utilities/cliUtils'
 import { isCloud9 } from '../shared/extensionUtilities'
 import { ToolkitError, isAwsError } from '../shared/errors'
 import { SsmClient } from '../shared/clients/ssmClient'
+import { showMessageWithUrl } from '../shared/utilities/messages'
+import { Ec2Client } from '../shared/clients/ec2Client'
+
+export type Ec2ConnectErrorName = 'permission' | 'instanceStatus'
+export interface Ec2ConnectErrorParameters {
+    message: string
+    url?: string
+    urlItem?: string
```

Comment:
maybe `url` could be added to `ErrorInformation` https://github.com/aws/aws-toolkit-vscode/blob/d28f9833e54cd518b898115b52d3ca19ec49efe1/src/shared/errors.ts#L20